### ISSUE TITLE
Why does NVTC need to calculate bit_u and bit_c during training, but in the code for generating binary streams, only bit_c has streams?

### ISSUE THREAD

**Issue by dellinehex7** [USTC-IMCL/NVTC#3]
Hi, thank you for the great work! 
I have some questions to consult with you. In the NVTC class in nvtc.py, why are bit_u and bit_c used simultaneously at non-lowest layers during training? I understand that bit_u should only be used at the lowest layer and not in CEM modules. Similarly, in the compress function of the NVTC class for generating binary streams, the streams at non-lowest layers do not include bit_u. Could you please explain the reasons for this? Thank you.

**Comment by RysonFeng** [comment]
Hi, @dellinehex7 ! Your understanding is general right! Except for the lowest layer, bits_u is mainly used for improving codeword usage during training. In fact, you can remove the bits_u[1:] item in loss function after the first 600000 training steps.

A  training issue for VQ based method is that the frequency of some codewords would be optimized to zero, and will be never used in testing.
This will lead to a performance drop and also a waste of computation resources.

Therefore, for each codebook with shape (cb_size, cb_dim), the model would maintain a logits table with shape (cb_size, 1). 
We can know how often a codeword is used, and reactivate the low-frequency codewords during training. 
Bits_u is then calculated for each codebook to optimize this (unconditional) logits.
